rclcpp_components/src/component_container_mt.cpp

```diff
@@ -22,8 +22,21 @@ int main(int argc, char * argv[])
 {
   /// Component container with a multi-threaded executor.
   rclcpp::init(argc, argv);
+
+  auto options = rclcpp::NodeOptions{}.allow_undeclared_parameters(true).
+  automatically_declare_parameters_from_overrides(true);
   auto exec = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
-  auto node = std::make_shared<rclcpp_components::ComponentManager>(exec);
+  auto node = std::make_shared<rclcpp_components::ComponentManager>(
+    exec, "ComponentManager", options);
+  if (node->has_parameter("thread_num")) {
```

Comment:
this parameter cannot be seen from other nodes including CLI, because `start_parameter_services` is disabled.

Reply:
that's not a problem though, you can still set it when running the component manager